src/rgw/rgw_rados.cc

```diff
@@ -6189,18 +6194,76 @@ int RGWRados::Object::Read::read(int64_t ofs, int64_t end, bufferlist& bl, optio
   return bl.length();
 }
 
-struct get_obj_data {
+
+/*
+
+struct get_obj_data : public RefCountedObject{
+  CephContext* cct; //amat
   RGWRados* store;
   RGWGetDataCB* client_cb;
+  RGWObjectCtx* ctx; //amat
+  librados::IoCtx io_ctx;
   rgw::Aio* aio;
   uint64_t offset; // next offset to write to client
+  uint64_t total_read; //amat
+  int sequence;
+
   rgw::AioResultList completed; // completed read results, sorted by offset
   optional_yield yield;
 
+  std::mutex lock;
+  std::mutex data_lock;
```

Comment:
thank you;  did these change later to ceph types?  those are a pain, I've been trying to see us use only std lock types since 2013

Reply:
yes, all locks in d3n code are std:: types